Formula/cgal.rb

```diff
@@ -21,12 +21,21 @@ def install
     args = std_cmake_args + %W[
       -DCMAKE_CXX_FLAGS='-std=c++14'
       -DWITH_Eigen3=ON
-      -DWITH_LAPACK=ON
       -DWITH_CGAL_Qt5=OFF
       -DWITH_CGAL_ImageIO=OFF
     ]
-
     system "cmake", ".", *args
+
+    # Manually edit the file that define CGAL_ROOT so it can work with cgal-qt5
+    # The goal is to make it work at least when cgal and cgal-qt5 are linked
+    rm buildpath/"config/CGALConfig-installation-dirs.cmake"
+    (buildpath/"config/CGALConfig-installation-dirs.cmake").write <<~EOS
+      if( EXISTS #{HOMEBREW_PREFIX}/include/CGAL/AABB_traits.h)
+        set(CGAL_ROOT #{HOMEBREW_PREFIX}/)
+      else()
+        set(CGAL_ROOT #{prefix})
+      endif()
+    EOS
```

Comment:
What is the point of this and why not just pick one or the other?

Reply:
This package is an extension of the CGAL library, and it uses the CGAL library to be built. The problem is then that it will be installed at some other place, as it is called cgal-qt5 and not cgal, but it needs the files of the cgal package. To do that, we use the HOMEBREW_PREFIX value to find CGAL if it has been linked, becaus ein that case, cgal and cgal-qt5 files are found in the same place. But in the case CGAL is not linked, we keep the real value so at least cgal works, even if cgal-qt5 doesn't.

Reply:
I'm not sure what it extends with but both formula need to be able to work separately. This is way too much magic.

A better solution might be to define both as conflicting and shipping an entire CGAL install in both.

Reply:
- @maxGimeno, can we implement that?
- @SMillerDev What would be the scenario for users? Would the installation of `cgal-qt5` automatically remove `cgal`, or would it emits an error?

Another solution would be to merge back `cgal-qt5` into the `cgal`, because do not add `depends_on "qt"`: that ways, the Qt5 support of CGAL would be available only if the user manually `brew install qt`.

Reply:
> Another solution would be to merge back cgal-qt5 into the cgal package, but do not add depends_on "qt": that way, the Qt5 support of CGAL would be available only if the user manually brew install qt.

That sounds great. It can just be a build dependency then.